wallet/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -538,15 +557,25 @@ getWalletPendingTxs wid =
 getPendingTx :: WebTypes.CId WebTypes.Wal -> TxId -> Query (Maybe PendingTx)
 getPendingTx wid txId = preview $ wsWalletInfos . ix wid . wsPendingTxs . ix txId
 
+-- | Legacy version of 'addCustomAddress2' for backwards compatibility on
+--   the event log.
+addCustomAddress :: CustomAddressType -> (WebTypes.CId WebTypes.Addr, HeaderHash) -> Update Bool
+addCustomAddress t (addr, hh) = fmap isJust $ customAddressL t . at (unsafeCIdToAddress addr) <<.= Just hh
+
 -- | If given address isn't yet present in set of used\/change addresses, then add it
 -- with given block header hash.
-addCustomAddress :: CustomAddressType -> (Address, HeaderHash) -> Update Bool
-addCustomAddress t (addr, hh) = fmap isJust $ customAddressL t . at addr <<.= Just hh
+addCustomAddress2 :: CustomAddressType -> (Address, HeaderHash) -> Update Bool
+addCustomAddress2 t (addr, hh) = fmap isJust $ customAddressL t . at addr <<.= Just hh
+
+-- | Legacy version of 'removeCustomAddress2' for backwards compatibility on
+--   the event log.
+removeCustomAddress :: CustomAddressType -> (WebTypes.CId WebTypes.Addr, HeaderHash) -> Update Bool
+removeCustomAddress t (cwa, hh) = removeCustomAddress2 t (unsafeCIdToAddress cwa, hh)
 
 -- | Remove given address from set of used\/change addresses only if provided
 -- header hash is equal to one which is stored in database.
-removeCustomAddress :: CustomAddressType -> (Address, HeaderHash) -> Update Bool
-removeCustomAddress t (addr, hh) = do
+removeCustomAddress2 :: CustomAddressType -> (Address, HeaderHash) -> Update Bool
+removeCustomAddress2 t (addr, hh) = do
```

Comment:
Instead of these transaction variants, lets add a newtype wrapper for `Address` that we define as a safe copy V1 type that migrates from `WebTypes.CId WebTypes.Addr`. Then the original transactions here should work as the input arguments will be migrated automatically.